core/templates/dev/head/pages/exploration_player/HintButtonDirective.js

```diff
@@ -33,11 +33,12 @@ oppia.directive('hintButton', [
         '/pages/exploration_player/' +
         'hint_button_directive.html'),
       controller: [
-        '$scope', '$rootScope', '$timeout', 'NumberAttemptsService',
+        '$scope', '$timeout', 'NumberAttemptsService',
         'WindowDimensionsService', 'TWO_CARD_THRESHOLD_PX',
-        function($scope, $rootScope, $timeout, NumberAttemptsService,
+        function($scope, $timeout, NumberAttemptsService,
             WindowDimensionsService, TWO_CARD_THRESHOLD_PX) {
           $scope.isShowingHintTooltip = false;
+          $scope.isWindowNarrow = WindowDimensionsService.isWindowNarrow();
```

Comment:
Can we leave the buttons as they are and do the ng-if in the parent container instead? Seems weird to go to all the trouble of instantiating a directive in the first place, just to have it be empty.

Also, consider binding to the function rather than evaluating it, i.e. `$scope.isWindowNarrow = WindowDimensionsService.isWindowNarrow;` and using `isWindowNarrow()` in the UI. Then check if resizing the window works correctly.